cudaextender/src/ungapped_xdrop.cu

```diff
@@ -0,0 +1,229 @@
+                                                                   d_query,
+                                                                   query_length,
+                                                                   d_score_mat_.data(),
+                                                                   no_entropy_,
+                                                                   xdrop_threshold_,
+                                                                   score_threshold,
+                                                                   d_seed_pairs,
+                                                                   curr_num_pairs,
+                                                                   seed_pair_start,
+                                                                   d_scored_segment_pairs,
+                                                                   d_done_.data());
+        size_t cub_storage_bytes = d_temp_storage_cub_.size();
+        GW_CU_CHECK_ERR(cub::DeviceScan::InclusiveSum(d_temp_storage_cub_.data(), cub_storage_bytes, d_done_.data(), d_done_.data(), curr_num_pairs, stream_))
+        // TODO- Make async
+        const int32_t num_scored_segment_pairs = get_value_from_device(d_done_.data() + curr_num_pairs - 1, stream_);
+        if (num_scored_segment_pairs > 0)
+        {
+            compress_output<<<1024, 1024, 0, stream_>>>(d_done_.data(),
+                                                        seed_pair_start,
+                                                        d_scored_segment_pairs,
+                                                        d_tmp_ssp_.data(),
+                                                        curr_num_pairs); // TODO- Need configurability for kernel?
+            thrust::stable_sort(thrust::cuda::par(allocator_).on(stream_),
+                                d_tmp_ssp_.begin(),
+                                d_tmp_ssp_.begin() + num_scored_segment_pairs,
+                                scored_segment_pair_comp());
+            GW_CU_CHECK_ERR(cub::DeviceSelect::Unique(d_temp_storage_cub_.data(),
+                                                      cub_storage_bytes,
+                                                      d_tmp_ssp_.data(),
+                                                      d_scored_segment_pairs + total_scored_segment_pairs_,
+                                                      d_num_scored_segment_pairs,
+                                                      num_scored_segment_pairs,
+                                                      stream_))
+            total_scored_segment_pairs_ += get_value_from_device(d_num_scored_segment_pairs, stream_);
+        }
+    }
+
+    set_device_value_async(d_num_scored_segment_pairs, &total_scored_segment_pairs_, stream_);
+
+    return StatusType::success;
+}
+
+StatusType UngappedXDrop::extend_async(const int8_t* h_query, const int32_t& query_length,
```

Comment:
Please change the `const int32_t&` to `const int32_t`.
Note: Here, in the implementation, it makes sense to use `const` if you want to ensure you don't accidentally change the value in the function. In the declaration I wouldn't use `const` on paramters that are passed by value, because the caller does not need to care if the copies remain constant in the function. The `const` just helps you as developer of the function.

Reply:
Done.

Reply:
Please make the pointers to read-only data `const type*` again. (See earlier comment.)

Reply:
Done.